hpy/devel/src/runtime/ctx_type.c

```diff
@@ -310,6 +317,36 @@ create_slot_defs(HPyType_Spec *hpyspec)
     return result;
 }
 
+static PyBufferProcs*
+create_buffer_procs(HPyType_Spec *hpyspec)
+{
+    PyBufferProcs *buffer_procs = NULL;
+    if (hpyspec->defines != NULL) {
+        for (int i = 0; hpyspec->defines[i] != NULL; i++) {
+            HPyDef *src = hpyspec->defines[i];
+            if (src->kind != HPyDef_Kind_Slot)
+                continue;
+            switch (src->slot.slot) {
+                case HPy_bf_getbuffer:
+                    if (buffer_procs == NULL) {
+                        buffer_procs = PyMem_Calloc(1, sizeof(PyBufferProcs));
```

Comment:
you should check that `PyMem_Calloc` didn't fail

Reply:
moreover, it's a bit sad that you need to duplicate the code for allocating `PyBufferProcs`. Maybe you can detect in advance whether it's needed by tweaking `HPyDef_count`?

Reply:
Error checking added.
I tried to find a way to factor out the allocation of PyBufferProcs, but everything I thought of just made things more complicated. Considering that there will only ever be 2 copies of that code, I'd rather leave it as is.